### PR TITLE
allow hotfixes to generate deb, rpm packages

### PR DESCRIPTION
**Pull request by harshavardhana**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
allow hotfixes to generate deb, rpm packages

## Motivation and Context
allow generating RPMs for the hotfixes given to customers

## How to test this PR?
Access to relevant credentials is needed

```
CRED_DIR=/media/harsha/minio make hotfix
```

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -142,13 +142,22 @@ hotfix-vars:
 	$(eval TAG := "minio/minio:$(VERSION)")
 
 hotfix: hotfix-vars install ## builds minio binary with hotfix tags
-	@mv -f ./minio ./minio.$(VERSION)
-	@minisign -qQSm ./minio.$(VERSION) -s "${CRED_DIR}/minisign.key" < "${CRED_DIR}/minisign-passphrase"
-	@sha256sum < ./minio.$(VERSION) | sed 's, -,minio.$(VERSION),g' > minio.$(VERSION).sha256sum
+	@wget -q -c https://github.com/minio/pkger/releases/download/v2.2.0/pkger_2.2.0_linux_amd64.deb
+	@wget -q -c https://raw.githubusercontent.com/minio/minio-service/v1.0.0/linux-systemd/distributed/minio.service
+	@sudo apt install ./pkger_2.2.0_linux_amd64.deb --yes
+	@mkdir -p minio-release/$(GOOS)-$(GOARCH)/archive
```
Comment (vadmeste) [inline Makefile:148]: it is better if we empty this directory so we don't end up with multiple hotfixes in our working directory

File: Makefile [hunk 2]
```diff
@@ -142,13 +142,22 @@ hotfix-vars:
...
+	@cp -af ./minio minio-release/$(GOOS)-$(GOARCH)/minio
+	@cp -af ./minio minio-release/$(GOOS)-$(GOARCH)/minio.$(VERSION)
+	@minisign -qQSm minio-release/$(GOOS)-$(GOARCH)/minio.$(VERSION) -s "${CRED_DIR}/minisign.key" < "${CRED_DIR}/minisign-passphrase"
+	@sha256sum < minio-release/$(GOOS)-$(GOARCH)/minio.$(VERSION) | sed 's, -,minio.$(VERSION),g' > minio-release/$(GOOS)-$(GOARCH)/minio.$(VERSION).sha256sum
+	@cp -af minio-release/$(GOOS)-$(GOARCH)/minio.$(VERSION)* minio-release/$(GOOS)-$(GOARCH)/archive/
+	@pkger -r $(VERSION) --ignore
```
Comment (vadmeste) [inline Makefile:154]: that are all needed arguments ? smart pkger :)

File: Makefile [hunk 3]
```diff
@@ -142,13 +142,22 @@ hotfix-vars:
...
 
 hotfix: hotfix-vars install ## builds minio binary with hotfix tags
-	@mv -f ./minio ./minio.$(VERSION)
-	@minisign -qQSm ./minio.$(VERSION) -s "${CRED_DIR}/minisign.key" < "${CRED_DIR}/minisign-passphrase"
-	@sha256sum < ./minio.$(VERSION) | sed 's, -,minio.$(VERSION),g' > minio.$(VERSION).sha256sum
+	@wget -q -c https://github.com/minio/pkger/releases/download/v2.2.0/pkger_2.2.0_linux_amd64.deb
+	@wget -q -c https://raw.githubusercontent.com/minio/minio-service/v1.0.0/linux-systemd/distributed/minio.service
+	@sudo apt install ./pkger_2.2.0_linux_amd64.deb --yes
+	@mkdir -p minio-release/$(GOOS)-$(GOARCH)/archive
+	@cp -af ./minio minio-release/$(GOOS)-$(GOARCH)/minio
```
Comment (vadmeste) [inline Makefile:149]: I think this is not needed

**Review by vadmeste — COMMENTED** [review]
Few comments

File: Makefile (hunk 3 above)
Comment (harshavardhana) [reply Makefile:149]: that is on purpose for the layout @vadmeste

File: Makefile (hunk 2 above)
Comment (harshavardhana) [reply Makefile:154]: yes @vadmeste

**Comment by harshavardhana** [comment]
updated PTAL @vadmeste

File: Makefile (hunk 3 above)
Comment (vadmeste) [reply Makefile:149]: this folder will contain multiples hotfixes binaries and it is confusing to have 'minio' binary in it; is this an historical problem ?

**Review by krisis — APPROVED** [review]

**Review by vadmeste — APPROVED** [review]
LGTM

**Pull request merged**